src/app/clusters/zone-management-server/zone-management-server.cpp

```diff
@@ -0,0 +1,767 @@
+{
+    // Perform constraint checks
+    if (HasFeature(Feature::kUserDefined))
+    {
+        VerifyOrReturnError(mMaxUserDefinedZones >= 5, CHIP_ERROR_INVALID_ARGUMENT,
+                            ChipLogError(Zcl, "ZoneManagement[ep=%d]: MaxUserDefinedZones configuration error", mEndpointId));
+        VerifyOrReturnError(mMaxZones >= mMaxUserDefinedZones, CHIP_ERROR_INVALID_ARGUMENT,
+                            ChipLogError(Zcl, "ZoneManagement[ep=%d]: MaxZones configuration error", mEndpointId));
+        int mfgCount = 0;
+        for (const auto & zone : mZones)
+        {
+            if (zone.zoneSource == ZoneSourceEnum::kMfg)
+            {
+                mfgCount++;
+            }
+        }
+        VerifyOrReturnError(mfgCount == (mMaxZones - mMaxUserDefinedZones), CHIP_ERROR_INVALID_ARGUMENT,
+                            ChipLogError(Zcl,
+                                         "ZoneManagement[ep=%d]: MfgZones + MaxUserDefinedZones must be equal to MaxZones count",
+                                         mEndpointId));
+    }
+    else
+    {
+        VerifyOrReturnError(mMaxZones >= 1, CHIP_ERROR_INVALID_ARGUMENT,
+                            ChipLogError(Zcl, "ZoneManagement[ep=%d]: MaxZones configuration error", mEndpointId));
+    }
+
+    VerifyOrReturnError(mSensitivityMax >= 2 && mSensitivityMax <= 10, CHIP_ERROR_INVALID_ARGUMENT,
+                        ChipLogError(Zcl, "ZoneManagement[ep=%d]: SensitivityMax configuration error", mEndpointId));
+
+    VerifyOrReturnError(AttributeAccessInterfaceRegistry::Instance().Register(this), CHIP_ERROR_INTERNAL);
+    ReturnErrorOnFailure(CommandHandlerInterfaceRegistry::Instance().RegisterCommandHandler(this));
+
+    return CHIP_NO_ERROR;
+}
+
+bool ZoneMgmtServer::HasFeature(Feature feature) const
+{
+    return mFeatures.Has(feature);
+}
+
+bool ZoneMgmtServer::SupportsOptAttr(OptionalAttribute aOptionalAttr) const
+{
+    return mOptionalAttrs.Has(aOptionalAttr);
+}
+
+CHIP_ERROR ZoneMgmtServer::ReadAndEncodeZones(const AttributeValueEncoder::ListEncodeHelper & encoder)
+{
+    for (const auto & zone : mZones)
+    {
+        ReturnErrorOnFailure(encoder.Encode(zone));
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ZoneMgmtServer::ReadAndEncodeTriggers(const AttributeValueEncoder::ListEncodeHelper & encoder)
+{
+    for (const auto & trigger : mTriggers)
+    {
+        ReturnErrorOnFailure(encoder.Encode(trigger));
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ZoneMgmtServer::SetSensitivity(uint8_t aSensitivity)
+{
+    VerifyOrReturnValue(aSensitivity != mSensitivity, CHIP_NO_ERROR);
+
+    if (aSensitivity < 1 || aSensitivity > mSensitivityMax)
+    {
+        return CHIP_IM_GLOBAL_STATUS(ConstraintError);
+    }
+
+    mDelegate.OnAttributeChanged(Attributes::Sensitivity::Id);
+    MatterReportingAttributeChangeCallback(ConcreteAttributePath(mEndpointId, ZoneManagement::Id, Attributes::Sensitivity::Id));
+
+    return CHIP_NO_ERROR;
```

Comment:
This is all great, but where is the actual set happening?

Was this tested at all?  If not, why not?

I am stopping here in the .cpp.

Reply:
It seems both the SetSensitivity() definition and call in Write() were added in commit `cb298cc8b53`, which was well before your review.

Reply:
Just for the record: the commit I reviewed did not have any changes to mSensitivity in Write() or anything called by Write().